In [1]:
!which python

/home/wangling/miniconda3/envs/p4ds-env/bin/python


# Final project: Phân tích dữ liệu Data Engineer Jobs
Các thành viên:

1. Phạm Thị Hoài Hiền - 18120178

2. Vương Thị Ngọc Linh - 18120195

In [2]:
import numpy as np 
import matplotlib as plt
import pandas as pd

## Giới thiệu về dữ liệu

Đây là bộ dữ liệu về việc làm Data Engineer và một số vị trí liên quan, link Kaggle lấy dữ liệu: https://www.kaggle.com/andrewmvd/data-engineer-jobs

Datatset này được tạo bởi Picklesueat và nó chứa hơn 2500 công việc liên quan vị trí Data Engineer.

Dataset gồm có 15 cột:
- Job Tile: Tên vị trí
- Salary Estimate: Mức lương
- Job Description: Miêu tả vị trí
- Rating: Xếp hạng công ty tuyển dụng
- Company name: Tên công ty tuyển dụng
- Location: Địa chỉ nơi làm việc
- Headquarters: Trụ sở của công ty
- Size: Quy mô nhân lực của công ty
- Founded: Năm thành lập công ty
- Type of ownership: Quyền sở hữu
- Industry: Lĩnh vực hoạt động của công ty
- Sector: Lĩnh vực chuyên ngành mà công ty phát triển.
- Revenue: Doanh thu mỗi năm của công ty.
- Competitors: Đối thủ cạnh tranh của công ty.
- Easy Apply: Vị trí này có dễ apply không.


## Tiền xử lí

** Đọc dữ liệu vào dataframe **

In [43]:
data_engineer_jobs = pd.read_csv("DataEngineer.csv")
data_engineer_jobs.shape

(2528, 15)

Dataset có 2528 dòng và 15 cột, đúng như miêu tả.

In [4]:
data_engineer_jobs.head()

Job Title  \
0                                      Data Engineer   
1  Senior Data Engineer (Healthcare Domain experi...   
2                                     Data Engineers   
3                      Client Trade Support Engineer   
4                                      Data Engineer   

               Salary Estimate  \
0  $80K-$150K (Glassdoor est.)   
1  $80K-$150K (Glassdoor est.)   
2  $80K-$150K (Glassdoor est.)   
3  $80K-$150K (Glassdoor est.)   
4  $80K-$150K (Glassdoor est.)   

                                     Job Description  Rating  \
0  Company Description\nSagence is a management a...     4.5   
1  Key Responsibilities\n\n- Architect, build, an...     3.4   
2  Overview\n\nJob description\n\nPosition Overvi...     5.0   
3  About the Position\n\n\nThis position will sup...     4.8   
4  Data Engineer\n\nJob Details\nLevel\nExperienc...     3.7   

                   Company Name      Location      Headquarters  \
0                  Sagence\n4.5  New York, NY       Chicago, IL   
1   Enterprise Integration\n3.4  New York, NY  Jacksonville, FL   
2     Maestro Technologies\n5.0  New York, NY       Trenton, NJ   
3              Jane Street\n4.8  New York, NY      New York, NY   
4  GNY Insurance Companies\n3.7  New York, NY      New York, NY   

                    Size  Founded  Type of ownership  \
0      1 to 50 employees     2009  Company - Private   
1    51 to 200 employees     1998  Company - Private   
2    51 to 200 employees     2003  Company - Private   
3  501 to 1000 employees     2000  Company - Private   
4   201 to 500 employees     1914  Company - Private   

                                Industry                  Sector  \
0                             Consulting       Business Services   
1                            IT Services  Information Technology   
2                            IT Services  Information Technology   
3  Investment Banking & Asset Management                 Finance   
4                     Insurance Carriers               Insurance   

                      Revenue                       Competitors Easy Apply  
0    $10 to $25 million (USD)               WCI Consulting, PwC         -1  
1    $25 to $50 million (USD)                                -1         -1  
2     $5 to $10 million (USD)                                -1         -1  
3    Unknown / Non-Applicable                                -1         -1  
4  $100 to $500 million (USD)  Travelers, Chubb, Crum & Forster       True

In [44]:
data_engineer_jobs = data_engineer_jobs.replace({"Uknown / Non-Applicable": np.NaN, "-1": np.NaN})

data_engineer_jobs

Job Title  \
0                                         Data Engineer   
1     Senior Data Engineer (Healthcare Domain experi...   
2                                        Data Engineers   
3                         Client Trade Support Engineer   
4                                         Data Engineer   
...                                                 ...   
2523                                     Cloud Engineer   
2524                                Electrical Engineer   
2525                              Sr. Big Data Engineer   
2526                     Senior Big Data / ETL Engineer   
2527                                     Cloud Engineer   

                  Salary Estimate  \
0     $80K-$150K (Glassdoor est.)   
1     $80K-$150K (Glassdoor est.)   
2     $80K-$150K (Glassdoor est.)   
3     $80K-$150K (Glassdoor est.)   
4     $80K-$150K (Glassdoor est.)   
...                           ...   
2523  $76K-$128K (Glassdoor est.)   
2524  $76K-$128K (Glassdoor est.)   
2525  $76K-$128K (Glassdoor est.)   
2526  $76K-$128K (Glassdoor est.)   
2527  $76K-$128K (Glassdoor est.)   

                                        Job Description  Rating  \
0     Company Description\nSagence is a management a...     4.5   
1     Key Responsibilities\n\n- Architect, build, an...     3.4   
2     Overview\n\nJob description\n\nPosition Overvi...     5.0   
3     About the Position\n\n\nThis position will sup...     4.8   
4     Data Engineer\n\nJob Details\nLevel\nExperienc...     3.7   
...                                                 ...     ...   
2523  The Expertise You Have Proven expertise in dev...     4.5   
2524  Lockheed Martin is seeking a Circuit and FPGA ...     2.9   
2525  Posting Title Principal Data Engineer (Contrac...     3.7   
2526  Your Opportunity\n\nCharles Schwab & Co., Inc ...     3.5   
2527  Cloud Engineer Westlake, TX 6+ months Descript...     3.6   

                                   Company Name        Location  \
0                                  Sagence\n4.5    New York, NY   
1                   Enterprise Integration\n3.4    New York, NY   
2                     Maestro Technologies\n5.0    New York, NY   
3                              Jane Street\n4.8    New York, NY   
4                  GNY Insurance Companies\n3.7    New York, NY   
...                                         ...             ...   
2523                            PMG Global\n4.5    Westlake, TX   
2524            Albin Engineering Services\n2.9  Fort Worth, TX   
2525  Cincinnati Bell Technology Solutions\n3.7      Irving, TX   
2526                                Schwab\n3.5    Westlake, TX   
2527        SPECTRAFORCE TECHNOLOGIES Inc.\n3.6    Westlake, TX   

             Headquarters                   Size  Founded  Type of ownership  \
0             Chicago, IL      1 to 50 employees     2009  Company - Private   
1        Jacksonville, FL    51 to 200 employees     1998  Company - Private   
2             Trenton, NJ    51 to 200 employees     2003  Company - Private   
3            New York, NY  501 to 1000 employees     2000  Company - Private   
4            New York, NY   201 to 500 employees     1914  Company - Private   
...                   ...                    ...      ...                ...   
2523          Herndon, VA      1 to 50 employees       -1  Company - Private   
2524      Santa Clara, CA    51 to 200 employees     1993  Company - Private   
2525       Cincinnati, OH  501 to 1000 employees       -1  Company - Private   
2526  Dettenheim, Germany                Unknown       -1   Company - Public   
2527          Raleigh, NC  501 to 1000 employees     2000  Company - Private   

                                   Industry                  Sector  \
0                                Consulting       Business Services   
1                               IT Services  Information Technology   
2                               IT Services  Information Technology   
3     Investment Banking & Asset Mana

** Tỉ lệ missing value của tưng cột sau khi chuyển các giá trị -1 và unknown -> NaN **

In [45]:
data_engineer_jobs.isnull().sum() / len(data_engineer_jobs) * 100

Job Title             0.000000
Salary Estimate       0.000000
Job Description       0.000000
Rating                0.000000
Company Name          0.000000
Location              0.000000
Headquarters          4.905063
Size                  4.707278
Founded               0.000000
Type of ownership     4.707278
Industry             14.952532
Sector               14.833861
Revenue               4.707278
Competitors          69.185127
Easy Apply           94.224684
dtype: float64

In [ ]:
Cột Competitors và Easy Apply có rất nhiều missing value, và dữ liệu hai cột này cũng không cần thiết cho câu hỏi, vì vậy sẽ xóa hai cột này để làm sạch dữ liệu.

In [46]:
data_engineer_jobs = data_engineer_jobs.drop(columns=["Competitors", "Easy Apply"])
data_engineer_jobs

Job Title  \
0                                         Data Engineer   
1     Senior Data Engineer (Healthcare Domain experi...   
2                                        Data Engineers   
3                         Client Trade Support Engineer   
4                                         Data Engineer   
...                                                 ...   
2523                                     Cloud Engineer   
2524                                Electrical Engineer   
2525                              Sr. Big Data Engineer   
2526                     Senior Big Data / ETL Engineer   
2527                                     Cloud Engineer   

                  Salary Estimate  \
0     $80K-$150K (Glassdoor est.)   
1     $80K-$150K (Glassdoor est.)   
2     $80K-$150K (Glassdoor est.)   
3     $80K-$150K (Glassdoor est.)   
4     $80K-$150K (Glassdoor est.)   
...                           ...   
2523  $76K-$128K (Glassdoor est.)   
2524  $76K-$128K (Glassdoor est.)   
2525  $76K-$128K (Glassdoor est.)   
2526  $76K-$128K (Glassdoor est.)   
2527  $76K-$128K (Glassdoor est.)   

                                        Job Description  Rating  \
0     Company Description\nSagence is a management a...     4.5   
1     Key Responsibilities\n\n- Architect, build, an...     3.4   
2     Overview\n\nJob description\n\nPosition Overvi...     5.0   
3     About the Position\n\n\nThis position will sup...     4.8   
4     Data Engineer\n\nJob Details\nLevel\nExperienc...     3.7   
...                                                 ...     ...   
2523  The Expertise You Have Proven expertise in dev...     4.5   
2524  Lockheed Martin is seeking a Circuit and FPGA ...     2.9   
2525  Posting Title Principal Data Engineer (Contrac...     3.7   
2526  Your Opportunity\n\nCharles Schwab & Co., Inc ...     3.5   
2527  Cloud Engineer Westlake, TX 6+ months Descript...     3.6   

                                   Company Name        Location  \
0                                  Sagence\n4.5    New York, NY   
1                   Enterprise Integration\n3.4    New York, NY   
2                     Maestro Technologies\n5.0    New York, NY   
3                              Jane Street\n4.8    New York, NY   
4                  GNY Insurance Companies\n3.7    New York, NY   
...                                         ...             ...   
2523                            PMG Global\n4.5    Westlake, TX   
2524            Albin Engineering Services\n2.9  Fort Worth, TX   
2525  Cincinnati Bell Technology Solutions\n3.7      Irving, TX   
2526                                Schwab\n3.5    Westlake, TX   
2527        SPECTRAFORCE TECHNOLOGIES Inc.\n3.6    Westlake, TX   

             Headquarters                   Size  Founded  Type of ownership  \
0             Chicago, IL      1 to 50 employees     2009  Company - Private   
1        Jacksonville, FL    51 to 200 employees     1998  Company - Private   
2             Trenton, NJ    51 to 200 employees     2003  Company - Private   
3            New York, NY  501 to 1000 employees     2000  Company - Private   
4            New York, NY   201 to 500 employees     1914  Company - Private   
...                   ...                    ...      ...                ...   
2523          Herndon, VA      1 to 50 employees       -1  Company - Private   
2524      Santa Clara, CA    51 to 200 employees     1993  Company - Private   
2525       Cincinnati, OH  501 to 1000 employees       -1  Company - Private   
2526  Dettenheim, Germany                Unknown       -1   Company - Public   
2527          Raleigh, NC  501 to 1000 employees     2000  Company - Private   

                                   Industry                  Sector  \
0                                Consulting       Business Services   
1                               IT Services  Information Technology   
2                               IT Services  Information Technology   
3     Investment Banking & Asset Mana

** Xem kiểu dữ liệu các cột **

In [47]:
data_engineer_jobs.dtypes

Job Title             object
Salary Estimate       object
Job Description       object
Rating               float64
Company Name          object
Location              object
Headquarters          object
Size                  object
Founded                int64
Type of ownership     object
Industry              object
Sector                object
Revenue               object
dtype: object



** Tiền xử lý các cột Salary Estimate, Revenue thành các cột có giá trị trung bình, kiểu dữ liệu số **


## Các câu hỏi cần trả lời

(chém gió vô đây)